### PR TITLE
Use macros for string literals and constants

### PR DESCRIPTION
**Pull request by Jinming-Hu**

### CODE REVIEW

File: sdk/storage/azure-storage-common/inc/azure/storage/common/constants.hpp
Comment (ahsonkhan) [inline sdk/storage/azure-storage-common/inc/azure/storage/common/constants.hpp:15]: Out of curiosity, what's motivating this change?

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-common/inc/azure/storage/common/constants.hpp:15]: is there anything wrong with macro?

Comment (ahsonkhan) [reply sdk/storage/azure-storage-common/inc/azure/storage/common/constants.hpp:15]: Not that I am aware off.

Comment (vhvb1989) [reply sdk/storage/azure-storage-common/inc/azure/storage/common/constants.hpp:15]: don't use MACRO,
those variables are not allocated in memory. Will they be always temporally allocated?  
Don't we pass strings by reference?   Then are you passing refs to temp objects?
https://rules.sonarsource.com/cpp/RSPEC-5028
https://stackoverflow.com/questions/42388077/constexpr-vs-macros

Why did you decide to make this change?
Can you get @CaseyCarter / @RickWinter  advise about this before merging?

Comment (CaseyCarter) [reply sdk/storage/azure-storage-common/inc/azure/storage/common/constants.hpp:15]: Macros are generally disliked due to lack of namespacing: your macro named `potato` stomps all over any other uses of the identifier `potato` in the program. These macros in particular are problematic in that they use names outside of the `ALL_CAPS_SNAKE_CASE` identifier space that is conventionally reserved for macros. I'd recommend avoiding macros for this use case, but if you must then I strongly suggest changing the names: we don't want to follow `<Windows.h>`'s example of defining `min`, `max`, and `small`.

**Review by RickWinter — CHANGES_REQUESTED** [review]
Using the macros could lead to negative side effects such as stomping on other values and/or being stomped on themself.  The previous solution was a better approach.

**Review by antkmsft — CHANGES_REQUESTED** [review]
1. Macro will be always visible to customer as non-namespaced `c_HttpHeaderXMsVersion`, which means they will never be able to use that name themselves. If we want macros, they should be at least prefixed and all-capsed: `AZ_STORAGE_HTTP_HEADER_XMS_VERSION`. BTW, what does `c_` prefix mean, regardless of macro or conastexpr? I don't think we have it defined in our guidelines. Constexprs in general would need to be named as simply PaascalCase: `HttpHeaderXMsVersion`

2. In general, guidelines prohibit macros, and all the exceptions would need to go through approval. General approach is to use non-macro means to achieve what you want, unless it is absolutely necessary to use macro, and there's simply not another way in the language to achieve what you want. Here, I don't think is the case - constexprs do allow you to have string literals. Are we missing context here? What problem does this change solves?

**Pull request closed without merging**
